pennylane/devices/qubit_mixed/sampling.py

```diff
@@ -133,12 +125,12 @@ def _measure_with_samples_diagonalizing_gates(
     total_indices = _get_num_wires(state, is_state_batched)
     wires = qml.wires.Wires(range(total_indices))
 
-    def _process_single_shot(samples):
+    def _process_single_shot_copy(samples):
         samples_processed = _process_samples(mp, samples, wires)
         if isinstance(mp, SampleMP):
             return math.squeeze(samples_processed)
         if isinstance(mp, CountsMP):
-            process_func = functools.partial(_process_counts_samples, mp_has_obs=mp.obs is not None)
+            process_func = functools.partial(mp.process_samples, wire_order=wires)
         elif isinstance(mp, ExpectationMP):
             process_func = _process_expval_samples
         elif isinstance(mp, VarianceMP):
```

Comment:
Can you also add a story to update this to use `mp.process_samples` for all measurements, not just counts?

Reply:
[sc-79761]